doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+convolved with a distinct kernel -- and the resulting feature maps will
+be summed up elementwise to produce the output feature map.
+
+The convolution depicted above is an instance of a 2-D convolution, but it can
+be generalized to N-D convolutions. For instance, in a 3-D convolution, the
+kernel would be a *cuboid* and would slide across the height, width and depth
+of the input feature map.
+
+The collection of kernels defining a discrete convolution has a shape
+corresponding to some permutation of :math:`(n, m, k_1, \ldots, k_N)`, where
+
+.. math::
+
+    \begin{split}
+        n &\equiv \text{number of output feature maps},\\
+        m &\equiv \text{number of input feature maps},\\
+        k_j &\equiv \text{kernel size along axis $j$}.
+    \end{split}
+
+The following properties affect the output size :math:`o_j` of a convolutional
+layer along axis :math:`j`:
+
+* :math:`i_j`: input size along axis :math:`j`,
+* :math:`k_j`: kernel size along axis :math:`j`,
+* :math:`s_j`: stride (distance between two consecutive positions of the
+  kernel) along axis :math:`j`,
+* :math:`p_j`: zero padding (number of zeros concatenated at the beginning and
+  at the end of an axis) along axis `j`.
+
+For instance, here is a :math:`3 \times 3` kernel applied to a
+:math:`5 \times 5` input padded with a :math:`1 \times 1` border of zeros using
+:math:`2 \times 2` strides:
+
+.. figure:: conv_arithmetic_figures/numerical_padding_strides.gif
+
+The analysis of the relationship between convolutional layer properties is eased
+by the fact that they don't interact across axes, i.e., the choice of kernel
+size, stride and zero padding along axis :math:`j` only affects the output size
+of axis :math:`j`. Because of that, this section will focus on the following
+simplified setting:
+
+* 2-D discrete convolutions (:math:`N = 2`),
+* square inputs (:math:`i_1 = i_2 = i`),
+* square kernel size (:math:`k_1 = k_2 = k`),
+* same strides along both axes (:math:`s_1 = s_2 = s`),
+* same zero padding along both axes (:math:`p_1 = p_2 = p`).
+
+This facilitates the analysis and the visualization, but keep in mind that the
+results outlined here also generalize to the N-D and non-square cases.
+
+Theano terminology
+==================
+
+Theano has its own terminology, which differs slightly from the convolution
+arithmetic guide's. Here's a simple conversion table for the two:
+
++------------------+-------------------------------------------------------------------+
+| Theano           | Convolution arithmetic                                            |
++==================+===================================================================+
+| ``filters``      | 4D collection of kernels                                          |
++------------------+-------------------------------------------------------------------+
+| ``input_shape``  | (batch size, input channels, :math:`i_1`, :math:`i_2`)            |
```

Comment:
I suggest to use:
(batch size (`b`), input channels (`c`), input rows (`i_1`), input columns `i_2`) 

so you can refer to the variable names later in the input_shape.